in-progress/12639-prd-inactivity-leak.md

```diff
@@ -0,0 +1,59 @@
+# Inactivity Leak Project Requirements Document
+
+- Owner: @spalladino
+- Approvers:
+  - @aminsammara
+  - @charlielye
+  - @PhilWindle
+  - @maddiaa
+  - @LHerskind
+
+## Background
+
+We currently have a slashing mechanism built on top of governance, where sequencers can vote to slash a node if agreed. Issue with this approach is that slashing takes time, and incentives (ie not wanting to lose money!) play a big role in it.
+
+However, during early testnet phases, where nodes do not have so much at stake, and failures are to be expected, we want to slash fast to ensure we don't halt block production due to a significant portion of validators being offline.
+
+So, for testnet only, we want to have a centralized entity who can rapidly slash or kick out validators from the set. This process can be manual, but we need to provide the slasher with accurate info on who to slash.
+
+## Desired User Flow
+
+The testnet Aztec Slasher (or AS for short, after their initials) must be able to gather all info needed for slashing from an Aztec Node under their control. Note that given slashing is centralized, we can use information from a single trusted node to decide who to slash.
+
+AS should be able to access this information either via a dashboard, or by hitting their node RPC interface and getting the data in JSON format. It must be immediately clear what nodes needs to be slashed from this data. For example:
+
+```
+$ curl -XPOST https://my-node/ -d'{"method": "node_getSlashable"}'
+{
+    "synchedTo": { "l1Block": 1, "l2Block": 1, "l2BlockHash": "0xabcd" } ,
+    "validators": [
+        { "address": "0x01", "lastBlockProposedAt": "2025-01-01T01:01:01", "lastBlockAttestedAt": "2025-01-01T01:01:01", "missedAttestationsStreak": 20, "missedProposalsStreak": 5, "missedProposalsRate": 0.1, "missedAttestationsRate": 0.2, "joinecAt": "2025-01-01T01:01:01" }
+    ]
+}
+```
+
+Once the addresses to slash have been gathered, AS, using a set of privileged L1 keys, should be able to call the L1 slasher contract to execute the slash in a single tx. This should be either executed through the Aztec CLI, or the Aztec CLI should generate the payload to run through cast. For example:
```

Comment:
Related to this. There are kinda two approaches; altering the slasher a bit and using that, or abusing that all staking assets are provided by us atm, and we can just "withdraw" them to exit people from the set.

Reply:
Always assuming testnet stuff has no value, of course